post-processor.go

```diff
@@ -84,16 +87,36 @@ func (p *PostProcessor) Configure(raws ...interface{}) error {
 	}
 
 	var cred *credentials.Credentials = nil // nil credentials use the default aws sdk credential chain
-	// Setting either credential config variable indicates an attempt to use configured credentials
-	if p.config.AccessKey != "" || p.config.SecretKey != "" {
+
+	if p.config.AccessKey != "" && p.config.SecretKey != "" {
+		// StaticProvider if both access id and secret are defined
+		// Environmental variables used:
+		// $AWS_SESSION_TOKEN
```

Comment:
Some of the comments here are now better explained in the README so IMO they could be removed from thw code. It's up to you though.